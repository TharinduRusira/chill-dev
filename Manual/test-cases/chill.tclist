
chill-testcase  test-cases/chill/test_distribute.script    test-cases/chill/mm.c  --check-run-script
chill-testcase  test-cases/chill/test_fuse.script          test-cases/chill/mm.c  --check-run-script
chill-testcase  test-cases/chill/test_known.script         test-cases/chill/mm.c  --check-run-script
chill-testcase  test-cases/chill/test_original.script      test-cases/chill/mm.c  --check-run-script
chill-testcase  test-cases/chill/test_peel.script          test-cases/chill/mm.c  --check-run-script
chill-testcase  test-cases/chill/test_permute.script       test-cases/chill/mm.c  --check-run-script
chill-testcase  test-cases/chill/test_print_code.script    test-cases/chill/mm.c  --check-run-script
chill-testcase  test-cases/chill/test_print_dep.script     test-cases/chill/mm.c  --check-run-script
chill-testcase  test-cases/chill/test_print_space.script   test-cases/chill/mm.c  --check-run-script
chill-testcase  test-cases/chill/test_reverse.script       test-cases/chill/mm.c  --check-run-script
chill-testcase  test-cases/chill/test_scale.script         test-cases/chill/mm.c  --check-run-script
chill-testcase  test-cases/chill/test_shift.script         test-cases/chill/mm.c  --check-run-script
chill-testcase  test-cases/chill/test_shift_to.script      test-cases/chill/mm.c  --check-run-script
chill-testcase  test-cases/chill/test_skew.script          test-cases/chill/mm.c  --check-run-script
chill-testcase  test-cases/chill/test_tile.script          test-cases/chill/mm.c  --check-run-script
chill-testcase  test-cases/chill/test_unroll_extra.script  test-cases/chill/mm.c  --check-run-script
chill-testcase  test-cases/chill/test_unroll.script        test-cases/chill/mm.c  --check-run-script

chill-testcase  test-cases/examples/chill/gemm.script             test-cases/examples/chill/gemm.c
chill-testcase  test-cases/examples/chill/gemv.script             test-cases/examples/chill/gemv.c
chill-testcase  test-cases/examples/chill/jacobi1.script          test-cases/examples/chill/jacobi1.c
chill-testcase  test-cases/examples/chill/jacobi2.script          test-cases/examples/chill/jacobi2.c
chill-testcase  test-cases/examples/chill/qr.script               test-cases/examples/chill/qr.c
#chill-testcase  test-cases/examples/chill/scalar_test.script      test-cases/examples/chill/scalar_test.c      --check-run-script --fail-run-script
chill-testcase  test-cases/examples/chill/swim.script             test-cases/examples/chill/swim.c
chill-testcase  test-cases/examples/chill/test_align.script       test-cases/examples/chill/test_align.c
#chill-testcase  test-cases/examples/chill/test_fusion.script      test-cases/examples/chill/test_fusion.c      --check-run-script --fail-run-script
#chill-testcase  test-cases/examples/chill/test_lex_order.script   test-cases/examples/chill/test_lex_order.c   --check-run-script --fail-run-script
chill-testcase  test-cases/examples/chill/test_split.script       test-cases/examples/chill/test_split.c
#chill-testcase  test-cases/examples/chill/test_split2.script      test-cases/examples/chill/test_split2.c      --check-run-script --fail-run-script
chill-testcase  test-cases/examples/chill/test_tile.script        test-cases/examples/chill/test_tile.c
#chill-testcase  test-cases/examples/chill/tile_violation.script   test-cases/examples/chill/tile_violation.c   --check-run-script --fail-run-script
chill-testcase  test-cases/examples/chill/unroll.script           test-cases/examples/chill/unroll.c
#chill-testcase  test-cases/examples/chill/unroll_violation.script test-cases/examples/chill/unroll_violation.c --check-run-script --fail-run-script

