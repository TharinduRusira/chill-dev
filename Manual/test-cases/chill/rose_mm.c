#define __rose_lt(x,y) ((x)<(y)?(x):(y))
#define __rose_gt(x,y) ((x)>(y)?(x):(y))

void mm(float **A,float **B,float **C,int ambn,int an,int bm)
{
  int t6;
  int t4;
  int t2;
  int chill_t12;
  int chill_t11;
  int chill_t10;
  int chill_t9;
  int chill_t8;
  int chill_t7;
  int chill_t6;
  int chill_t5;
  int chill_t4;
  int chill_t3;
  int chill_t2;
  int chill_t1;
  int i;
  int j;
  int n;
  for (t2 = 0; t2 <= an - 1; t2 += 1) 
    for (t4 = 0; t4 <= bm - 1; t4 += 1) {
      C[t2][t4] = 0.0f;
      C[t2][t4] += (A[t2][0] * B[0][t4]);
      C[t2][t4] += (A[t2][1] * B[1][t4]);
      C[t2][t4] += (A[t2][2] * B[2][t4]);
      C[t2][t4] += (A[t2][3] * B[3][t4]);
      for (t6 = 4; t6 <= ambn - 1; t6 += 1) 
        C[t2][t4] += (A[t2][t6] * B[t6][t4]);
    }
}
