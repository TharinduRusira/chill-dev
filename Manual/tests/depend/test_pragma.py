from chill import *

source('mm.c')
procedure('mm')
loop(0)
permute([3,1,2])
print_code()
print_space()


print("num statements = ", num_statements())
print_dep()
