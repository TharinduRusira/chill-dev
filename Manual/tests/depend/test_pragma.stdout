if (bm >= 1) {
  if (an >= 1) {
    for(t2 = 0; t2 <= ambn-1; t2++) {
      for(t4 = 0; t4 <= an-1; t4++) {
        if (t2 <= 0) {
          for(t6 = 0; t6 <= bm-1; t6++) {
            s0(t4,t6,t2);
            s1(t4,t6,t2);
          }
        }
        else {
          for(t6 = 0; t6 <= bm-1; t6++) {
            s1(t4,t6,t2);
          }
        }
      }
    }
  }
  if (ambn <= 0) {
    for(t4 = 0; t4 <= an-1; t4++) {
      for(t6 = 0; t6 <= bm-1; t6++) {
        s0(t4,t6,0);
      }
    }
  }
}


s0: { Sym=[bm,an] [t1,t2,t3,t4,t5,t6,t7] : t7 = 0 && t3 = 0 && t5 = 0 && t1 = 0 && t2 = 0 && 0 <= t4 < an && 0 <= t6 < bm }
s1: { Sym=[ambn,bm,an] [t1,t2,t3,t4,t5,t6,t7] : t7 = 1 && t3 = 0 && t5 = 0 && t1 = 0 && 0 <= t4 < an && 0 <= t6 < bm && 0 <= t2 < ambn }
('num statements = ', 2)
dependence graph:
s0->s1: C:true(+, 0, 0) C:true(0, 0, 0) C:output(+, 0, 0) C:output(0, 0, 0)
s1->s1: C:anti(+, 0, 0) C:output(+, 0, 0)
